spec/requests/user_data_spec.rb

```diff
@@ -157,4 +156,48 @@
     end
   end
 
+  describe 'request error messages' do
+    describe 'exception TokenNotValidError raised' do
```

Comment:
Although the outer `describe` is fine (line 159) each of these inner `describe` blocks should be `context` blocks, as with each one you are setting up a *context* in which new conditions apply

(a good rule of thumb is if you have a `before` or `let(...) blocks, it should probably be a context. If it's purely describing something, use `describe`)